src/key-casing.ts

```diff
@@ -38,7 +38,26 @@ function deepTransformKeys<T>(obj: T, transform: (s: string) => string): T {
 }
 
 /**
- * Transforms the keys of an Record to camelCase.
+ * This function is used to shallowly transform the keys of an object.
+ * It will only be transformed at runtime, so it's not type safe.
+ * @param obj the object to transform.
+ * @param transform the function to transform the keys from string to string.
+ * @returns the transformed object.
+ * @example transformKeys({ 'foo-bar': { 'fizz-buzz': true } }, toCamelCase)
+ * // { fooBar: { 'fizz-buzz': true } }
+ */
+function transformKeys<T>(obj: T, transform: (s: string) => string): T {
+  if (typeOf(obj) !== 'object') return obj
```

Comment:
no it won't because it's the `typeOf` function from the internals and not the standard `typeof` one (similar to the `deepTransformKeys` function above).
Maybe I can add a test though?